avocado/core/jobdata.py

```diff
@@ -0,0 +1,217 @@
+
+
+def retrieve_urls(resultsdir):
+    """
+    Retrieves the job urls from the results directory.
+    """
+    recorded_urls = os.path.join(resultsdir, JOB_DATA_DIR, URLS_FILENAME)
+    if not os.path.exists(recorded_urls):
+        recorded_urls = os.path.join(resultsdir,
+                                     JOB_DATA_FALLBACK_DIR,
+                                     URLS_FILENAME)
+        if not os.path.exists(recorded_urls):
+            return None
+
+    with open(recorded_urls, 'r') as urls_file:
+        urls = urls_file.read()
+
+    return ast.literal_eval(urls)
+
+
+def retrieve_mux(resultsdir):
+    """
+    Retrieves the job multiplex from the results directory.
+    """
+    recorded_mux = os.path.join(resultsdir, JOB_DATA_DIR, MUX_FILENAME)
+    if not os.path.exists(recorded_mux):
+        recorded_mux = os.path.join(resultsdir,
+                                    JOB_DATA_FALLBACK_DIR,
+                                    MUX_FILENAME)
+        if not os.path.exists(recorded_mux):
+            return None
+
+    with open(recorded_mux, 'r') as mux_file:
+        return pickle.load(mux_file)
+
+
+def retrieve_args(resultsdir):
+    """
+    Retrieves the job args from the results directory.
+    """
+    recorded_args = os.path.join(resultsdir, JOB_DATA_DIR, ARGS_FILENAME)
+    if not os.path.exists(recorded_args):
+        recorded_args = os.path.join(resultsdir,
+                                     JOB_DATA_FALLBACK_DIR,
+                                     ARGS_FILENAME)
+        if not os.path.exists(recorded_args):
+            return None
+
+    with open(recorded_args, 'r') as args_file:
+        return pickle.load(args_file)
+
+
+def retrieve_config(resultsdir):
+    """
+    Retrieves the job settings from the results directory.
+    """
+    recorded_config = os.path.join(resultsdir, JOB_DATA_DIR, CONFIG_FILENAME)
+    if not os.path.exists(recorded_config):
+        recorded_config = os.path.join(resultsdir,
+                                       JOB_DATA_FALLBACK_DIR,
+                                       CONFIG_FILENAME)
+        if not os.path.exists(recorded_config):
+            return None
+
+    return recorded_config
+
+
+def retrieve_cmdline(resultsdir):
+    """
+    Retrieves the job command line from the results directory.
+    """
+    recorded_cmdline = os.path.join(resultsdir, JOB_DATA_DIR, CMDLINE_FILENAME)
+    if not os.path.exists(recorded_cmdline):
+        recorded_cmdline = os.path.join(resultsdir,
+                                        JOB_DATA_FALLBACK_DIR,
+                                        CMDLINE_FILENAME)
+        return None
+
+    with open(recorded_cmdline, 'r') as cmdline_file:
+        return ast.literal_eval(cmdline_file.read())
+
+
+def get_resultsdir(logdir, jobid):
+    """
+    Gets the job results directory using a Job ID.
+    """
+    if jobid == 'latest':
+        try:
+            actual_dir = os.readlink(os.path.join(logdir, 'latest'))
```

Comment:
This works now, but we might some day use non-relative links to latest and this will be suddenly broken. It should be `os.realpath(os.path.join(logdir, 'latest'))` instead. Also I'm not sure why do you do this as `replay` should work with any directory containing `id` so this could simply just check whether the directory exists (and we'd get support for other existing directories inside the logdir).

Anyway my concern is only the first part, which is that this is potential dangerous and can return incorrect path.